Sources/Brave/Frontend/Browser/Helpers/LaunchHelper.swift

```diff
@@ -127,6 +123,24 @@ public actor LaunchHelper {
   }
 }
 
+private extension FilterListStorage {
+  var validBlocklistTypes: Set<ContentBlockerManager.BlocklistType> {
+    if filterLists.isEmpty {
+      return allFilterListSettings.reduce(Set<ContentBlockerManager.BlocklistType>()) { partialResult, setting in
+        guard let componentId = setting.componentId else { return partialResult }
+        return partialResult.union([
+          .filterList(componentId: componentId, isAlwaysAggressive: setting.isAlwaysAggressive)
+        ])
+      }
+    } else {
+      return filterLists.reduce(Set<ContentBlockerManager.BlocklistType>()) { partialResult, filterList in
+        return partialResult.union([
+          .filterList(componentId: filterList.entry.componentId, isAlwaysAggressive: filterList.isAlwaysAggressive)
+        ])
+      }
+    }
+  }
+}
```

Comment:
Here I realized I was unnecessarily removing content blockers when filter lists were not yet ready.